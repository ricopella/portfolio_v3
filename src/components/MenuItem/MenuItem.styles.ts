import styled from '@emotion/styled'
import { AnchorLink } from 'gatsby-plugin-anchor-links'
import { COLORS, DIMENSIONS, FONT_WIDTHS } from '../../styles/variables'
import { motion } from 'framer-motion'

const MenuItemLi = styled(motion.li)`
  align-items: center;
  cursor: pointer;
  display: flex;
  list-style: none;
  margin-bottom: 1.25rem;
  margin: 0;
  padding: 0;

  &:hover {
    cursor: none;
  }

  &.active {
    pointer-events: all;

    &:hover {
      cursor: pointer;
    }
  }
`

const HomepageLink = styled(AnchorLink)`
  color: ${props => props.theme.actionColor};
  cursor: pointer;
  font-size: ${DIMENSIONS.fontSize.regular};
  font-weight: ${FONT_WIDTHS.SEMI_BOLD};
  align-self: center;
  justify-self: center;
  text-transform: capitalize;

  &:hover,
  &:focus {
    text-decoration: underline;
  }
`

const MENU_ITEM_VARIANTS = {
  open: {
    y: 0,
    opacity: 1,
    display: 'initial',
    transition: {
      y: { stiffness: 1000, velocity: -100 },
    },
  },
  closed: {
    y: 50,
    opacity: 0,
    display: 'none',
    transition: {
      y: { stiffness: 1000 },
    },
  },
}

export default {
  MenuItemLi,
  HomepageLink,
  MENU_ITEM_VARIANTS,
}
