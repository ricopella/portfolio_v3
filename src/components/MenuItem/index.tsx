import React, { FC } from 'react'
import setHref from '../../utils/setHref'
import Styled from './MenuItem.styles'
import { BUTTON_ANIMATION } from '../../styles/variables'
import { MenuItemProps } from '../../types'

/**
 * Component used for Header and Navigation
 *
 * Since the animation requires
 *
 * @param {*} { isOpen, item }
 */
const MenuItem: FC<MenuItemProps> = ({ item }) => (
  <Styled.MenuItemLi variants={Styled.MENU_ITEM_VARIANTS} {...BUTTON_ANIMATION}>
    {item === 'resume' ? (
      <Styled.ResumeLink to="/resume">+ {item}</Styled.ResumeLink>
    ) : (
      <Styled.HomepageLink
        to={setHref(item)}
      >{`+ ${item}`}</Styled.HomepageLink>
    )}
  </Styled.MenuItemLi>
)

export default MenuItem
