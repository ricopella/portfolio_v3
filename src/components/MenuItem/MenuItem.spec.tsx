import MenuItem from './index'
import React from 'react'
import renderer from 'react-test-renderer'

describe('<MenuItem />', () => {
  const props = {
    item: 'foo',
  }

  it('should match MenuItem snapshot', () => {
    // @ts-ignore
    const internalTree = renderer.create(<MenuItem {...props} />).toJSON()

    expect(internalTree).toMatchSnapshot()
  })
})
