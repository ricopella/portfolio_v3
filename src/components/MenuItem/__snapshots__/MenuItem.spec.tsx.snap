// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> should match MenuItem snapshot 1`] = `
<li
  className="css-wimf6w"
  style={Object {}}
>
  <a
    className="css-1cmy6pj"
    href="/#foo"
    onClick={[Function]}
  >
    + foo
  </a>
</li>
`;
